### PR TITLE
Port all entities to use new `EntityObjectLoaderRtkQuery`

### LINKED ISSUE #50323: Port all entities to use new `EntityObjectLoaderRtkQuery`
Depends on #44491.

In #44491 only 1 or 2 types of entities should have been ported.
The goal of this issue is to port all remaining entities to use new `EntityObjectLoaderRtkQuery`.

⚠  Search for issue URL in code for TODOs.

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #50323

### Description

#44523 enabled the new `EntityObjectLoaderRtkQuery` only for Databases and Tables.
This PR enables it for the rest of entities (except Actions - will be done separately in #50873).

This PR may break data fetching (if e2e tests didn't cover a given case) and may increase probability of flakes.

### How to verify

CI is green.

### CODE REVIEW

File: frontend/src/metabase/api/tags/utils.ts [hunk 1]
```diff
@@ -364,9 +365,14 @@ export function providePermissionsGroupListTags(
 }
 
 export function providePermissionsGroupTags(
-  group: GroupListQuery,
+  group: Group | GroupListQuery,
 ): TagDescription<TagType>[] {
-  return [idTag("permissions-group", group.id)];
+  return [
```
Comment (kamilmielnik) [inline frontend/src/metabase/api/tags/utils.ts:370]: We want to refresh permissions group when a user in that group is e.g. deleted

File: frontend/src/metabase/containers/SavedQuestionLoader.jsx [hunk 2]
```diff
@@ -54,23 +53,16 @@ const SavedQuestionLoader = ({
     await dispatch(loadMetadataForCard(loadedQuestion.card()));
   }, [loadedQuestion?.id()]);
 
-  useEffect(() => {
-    if (loadedQuestion?.id() == null) {
-      return;
-    }
-
+  const question = useMemo(() => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/containers/SavedQuestionLoader.jsx:56]: The `useMemo` is semantically equivalent to previous `useState` + `useEffect` but it helps avoid a re-render. There was a case where this extra intermediate state (in-between `setQuestion` call and the re-render caused by it) caused things to crash down the tree.

File: frontend/src/metabase/entities/fields.js [hunk 3]
```diff
@@ -221,4 +241,32 @@ const Fields = createEntity({
...
+
+  const { data } = result;
+  const transformedData = useMemo(() => {
+    return data ? transformFieldValuesData(data, tableId) : data;
+  }, [data, tableId]);
+
+  return useMemo(
+    () => ({ ...result, data: transformedData }),
+    [result, transformedData],
+  );
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/fields.js:258]: All that to mimic this: https://github.com/metabase/metabase/blob/f1607ca2138e944516ec4c9e4f91448c84715a06/frontend/src/metabase/entities/fields.js#L117-L125

File: frontend/src/metabase/entities/schemas.js [hunk 4]
```diff
@@ -180,3 +193,42 @@ function addTableAvoidingDuplicates(tables, tableId) {
...
+
+  const virtualDatabaseTables = useListVirtualDatabaseTablesQuery(
+    opts?.isDatasets ? query : skipToken,
+  );
+
+  const databaseSchemaTables = useListDatabaseSchemaTablesQuery(
+    opts?.isDatasets ? skipToken : query,
+  );
+
+  const tables = opts?.isDatasets
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/schemas.js:214]: Conditional call, to mimic this:
https://github.com/metabase/metabase/blob/ba3b7818e6cdf155a74bf00f2b3310c9eec616fc/frontend/src/metabase/entities/schemas.js#L64-L90

File: frontend/src/metabase/entities/schemas.js [hunk 5]
```diff
@@ -180,3 +193,42 @@ function addTableAvoidingDuplicates(tables, tableId) {
   }
   return tables.includes(tableId) ? tables : [...tables, tableId];
 }
+
+const useGetQuery = ({ id, ...args }) => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/schemas.js:197]: This hook mimics this: https://github.com/metabase/metabase/blob/ba3b7818e6cdf155a74bf00f2b3310c9eec616fc/frontend/src/metabase/entities/schemas.js#L64-L90

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataFieldSettings/MetadataFieldSettings.tsx [hunk 6]
```diff
@@ -208,6 +208,15 @@ const FieldBreadcrumbs = ({
 
 // eslint-disable-next-line import/no-default-export -- deprecated usage
 export default _.compose(
+  Fields.load({
```
Comment (kamilmielnik) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataFieldSettings/MetadataFieldSettings.tsx:211]: This fixes a crash.
The order matters here.

File: frontend/src/metabase/collections/components/CollectionContent/CollectionContent.tsx [hunk 7]
```diff
@@ -36,7 +38,7 @@ export function CollectionContent({
       "exclude-archived": true,
     });
 
-  const { data: collection, error: collectionError } = useCollectionQuery({
+  const { data: collection, error: collectionError } = useGetCollectionQuery({
```
Comment (kamilmielnik) [inline frontend/src/metabase/collections/components/CollectionContent/CollectionContent.tsx:41]: There were 2 cases with Collections where hooks from `hooks/entity-framework` would clash with RTK query somehow, and RTK query would end up in forever-loading state. I'm not entirely sure why, but I'm sure replacing the old hooks with RTK query hooks helps.

I think it'd be safest to do this with all entities, so I created a separate issue for it and added it to the epic: #50873

File: frontend/src/metabase/common/components/CollectionPicker/components/CollectionPicker.tsx [hunk 8]
```diff
@@ -60,14 +60,12 @@ export const CollectionPickerInner = (
...
-
+  const id = isValidCollectionId(initialValue?.id) ? initialValue?.id : "root";
   const {
     data: currentCollection,
     error,
     isLoading: loadingCurrentCollection,
-  } = useCollectionQuery({
-    id: isValidCollectionId(initialValue?.id) ? initialValue?.id : "root",
-  });
+  } = useGetCollectionQuery(id != null ? { id } : skipToken);
```
Comment (kamilmielnik) [inline frontend/src/metabase/common/components/CollectionPicker/components/CollectionPicker.tsx:68]: There were 2 cases with Collections where hooks from `hooks/entity-framework` would clash with RTK query somehow, and RTK query would end up in forever-loading state. I'm not entirely sure why, but I'm sure replacing the old hooks with RTK query hooks helps.

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataFieldSettings/MetadataFieldSettings.tsx (hunk 6 above)
Comment (uladzimirdev) [reply frontend/src/metabase/admin/datamodel/metadata/components/MetadataFieldSettings/MetadataFieldSettings.tsx:211]: what kind of crash?

File: frontend/src/metabase/api/tags/utils.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/api/tags/utils.ts:383]: This could/should be refactored so there is no union and `in`

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/api/tags/utils.ts (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase/api/tags/utils.ts:383]: e3bb35ce68c3555b96784c6965c1be1907f9e805

Comment (ranquild) [reply frontend/src/metabase/api/tags/utils.ts:383]: lgtm

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataFieldSettings/MetadataFieldSettings.tsx (hunk 6 above)
Comment (kamilmielnik) [reply frontend/src/metabase/admin/datamodel/metadata/components/MetadataFieldSettings/MetadataFieldSettings.tsx:211]: Reading property of an undefined object. I didn't find the root cause.
It made 1 test fail (I can't remember which one, but it's easy to find out :)).

**Pull request merged**